semgrep/setup.py

```diff
@@ -68,6 +69,9 @@ class PostInstallCommand(install):
     """Post-installation for installation mode."""
 
     def run(self):
+        if platform.system() == "Windows":
+            print("Not attempting to install binary while running under windows")
+            return
```

Comment:
I think it's better to simply fail the installation altogether on Windows. As I understand it, we're not currently compiling a version of `semgrep-core` for Windows, nor are we providing instructions for user's to do it themselves. So even if someone successfully installs the Python package on Windows they won't be able to do anything useful since they'll be missing `semgrep-core`.

What do you think @brendongo?

Reply:
@mschwager Does this failure affect the installation of other libraries that are cross-platform and uses semgrep. How about a warning message that Windows is not supported and a graceful exit?

Reply:
If some project A depends on semgrep as a dependency and someone installs A on windows, does this graceful exit let them know semgrep-core was not installed? I guess A can check if semgrep-core doesn't exist and bypass running semgrep rules?

Reply:
This is exactly what I do 

Do not run semgrep on windows: https://github.com/ajinabraham/libsast/blob/master/libsast/core_sgrep/helpers.py#L11
and in setup.py: https://github.com/ajinabraham/libsast/blob/master/setup.py#L13

But I still see that semgrep installation is attempted on windows.

https://travis-ci.com/github/MobSF/Mobile-Security-Framework-MobSF/jobs/366333490#L523

I am not sure if this is because the libsast wheel https://pypi.org/project/libsast/#files in pypi is generated from Mac.

Reply:
It looks like semgrep is being added to the requires.txt of libsast 1.0.8 which I think is installed before setup.py code.

Reply:
Okay, I have a few suggestions here. It looks like we have two problems: 1.) Semgrep's `setup.py` is trying to install on Windows when it should be failing early and loudly, and 2.) an upstream dependent of `libsast` is getting build errors on Windows machines.

Regarding (1), I like your solution here @ajinabraham and I appreciate you helping us out in this respect. My one suggestion would be to pull this check out to the global scope. That way we fail as early as possible, not once we've made it to `PostInstallCommand`. I'm envisioning something like:

```python
if platform.system() == "Windows":
    print("Semgrep does not currently support Windows")
    sys.exit(1)
```

Regarding (2), I found these useful docs on [declaring platform specific dependencies](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies). Would it be possible to avoid your `platform.system() != 'Windows'` check and instead unconditionally require `semgrep==0.16.0;platform_system!='Windows'`. That way, even when you build the package on macOS or Linux it will still avoid installing `semgrep` on Windows. What do you think?

Reply:
Nice! I can make it work with `Option 2`. Thanks for digging the docs. This should allow me to include semgrep without barfing errors on Windows.

I will close this PR for now as I hope you will support Windows in the long term.